apps/cowswap-frontend/src/legacy/utils/trade.ts

```diff
@@ -129,7 +135,7 @@ export function getSignOrderParams(params: PostOrderParams): SignOrderParams {
       buyAmount,
       validTo,
       appData: appData.appDataKeccak256,
-      feeAmount: '0',
+      feeAmount: (swapZeroFee ? '0' : feeAmount?.quotient.toString()) || '0',
```

Comment:
It defaults to the behavior matching when the feature flag is on?

Reply:
It's actually unreal case when `feeAmount` is falsy. This is a part of legacy code